### PR TITLE
232 adds an affine transform

### PR DESCRIPTION
**Pull request by wyli**
subtask of #232, revisit #270 as PR270 is out-of-date
adds an affine transform module, which can be configured to
- be a part of the spatial transformer network (tested against the implementation from https://pytorch.org/tutorials/intermediate/spatial_transformer_tutorial.html)
- partly replicate the result of scipy.ndimage.affine_transform using pytorch (tested against cases generated with scipy; prefiltering not implemented, pytorch resampler currently supports linear interpolation)

this PR also fixes a few minor documentation issues

### Status
**Ready**

### Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New tests added to cover the changes
- [x] Docstrings/Documentation updated

### CODE REVIEW

File: monai/transforms/spatial/dictionary.py [hunk 1]
```diff
@@ -514,7 +514,9 @@ def __call__(self, data):
             device = self.rand_3d_elastic.device
             grid = torch.tensor(grid).to(device)
             gaussian = GaussianFilter(spatial_dims=3, sigma=self.rand_3d_elastic.sigma, truncated=3.0).to(device)
-            grid[:3] += gaussian(self.rand_3d_elastic.rand_offset[None])[0] * self.rand_3d_elastic.magnitude
+            grid[:3] += (
+                gaussian(torch.tensor(self.rand_3d_elastic.rand_offset[None]))[0] * self.rand_3d_elastic.magnitude
```
[the remark this answers is not in the document]
Comment (wyli) [reply monai/transforms/spatial/dictionary.py:518]: good point, thanks, I'll change this. 

to be consistent with pytorch nn.modules, I think monai.networks.layers should assume that the input tensors have the correct device and contiguous properties. layer users (in this case it's the transform) are responsible for preparing these properties

**Comment by wyli** [comment]
FYI @edgarriba I'm sure this could be replaced with a Kornia implementation in the future

**Comment by edgarriba** [comment]
@wyli thanks. I'll push this week a first version for `warp_persepctive` which the header looks something like this:
```python
def warp_projective(src: torch.Tensor,
                    P: torch.Tensor,
                    dsize: Tuple[int, int, int],
                    flags: str = 'bilinear',
                    padding_mode: str = 'zeros',
                    align_corners: bool = True) -> torch.Tensor:
    r"""Applies a projective transformation a to 3d tensor.

        src (torch.Tensor): input tensor of shape :math:`(B, C, D, H, W)`.
        P (torch.Tensor): projective transformation of shape :math:`(B, 3, 4)`.
        dsize (Tuple[int, int, int]): size of the output image (depth, height, width).
        mode (str): interpolation mode to calculate output values
          'bilinear' | 'nearest'. Default: 'bilinear'.
        padding_mode (str): padding mode for outside grid values
          'zeros' | 'border' | 'reflection'. Default: 'zeros'.
        align_corners (bool): mode for grid_generation. Default: True. See
        https://pytorch.org/docs/stable/nn.functional.html#torch.nn.functional.interpolate for details

    Returns:
        torch.Tensor: the warped 3d tensor with shape :math:`(B, C, D, H, W)`.

    """
```

have also a couple of helper methods to produce the transforms based on a given angle axis vector and the origin where to apply such transformation

```python

def get_projection_matrix3d(center: torch.Tensor, angles: torch.Tensor) -> torch.Tensor:
    r"""Calculates the projection matrix for a 3D rotation.

    The function computes the projection matrix given the center, scales and angles per axis.

    Args:
        center (torch.Tensor): center of the rotation in the source with shape :math:`(B, 3)`.
        angles (Tensor): angle axis vector containing the rotation angles in degrees in the form
            of (rx, ry, rx) with shape :math:`(B, 3)`. Internally it calls Rodrigues to compute
            the rotation matrix from axis-angle.

    Returns:
        Tensor: the projection matrix of 3D rotation with shape :math:`(B, 3, 4)`.

    """```
